### DISCUSSION TITLE
Plugin Development Help: Entity metadata for ThreesixtyView#product was not found.

### DISCUSSION THREAD

**Discussion by sheikhumar93** [medusajs/medusa#4635] [Q&A]
I am trying to develop a plugin, I have created a new Entity ThreesixtyView in the plugin which needs to have a many-to-one relationship with the default Product Entity

```
import { DbAwareColumn, BaseEntity, generateEntityId } from "@medusajs/medusa";
import { Product } from "@medusajs/medusa";

@Entity({ name: "threesixty_view" })
export class ThreesixtyView extends BaseEntity {
  @Column({ type: "varchar", nullable: false })
  product_id: string;

  @ManyToOne(() => Product)
  @JoinColumn({ name: "product_id" })
  product: Product;

  @DbAwareColumn({ type: "jsonb", nullable: true })
  metadata: Record<string, unknown>;

  @BeforeInsert()
  private beforeInsert(): void {
    this.id = generateEntityId(this.id, "threesixtyview");
  }
}
```

After running `yarn build` in my plugin folder, when I use `medusa develop` in my main backend folder, an error is thrown 

`TypeORMError: Entity metadata for ThreesixtyView#product was not found. Check if you specified a correct entity object and if it's connected in the connection options.
`

When I comment out `product: Product` in my new entity it starts working again. I have checked that the relationship exists in the Database as I have created the migration for it and the migration runs without any issue.

```
public async up(queryRunner: QueryRunner): Promise<void> {
    await queryRunner.query(
      `CREATE TABLE "threesixty_view" ("id" character varying NOT NULL, "created_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "updated_at" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "deleted_at" TIMESTAMP WITH TIME ZONE, "metadata" jsonb, "product_id" character varying NOT NULL, CONSTRAINT "PK_c5cb94be2d8011eebe560242ac1" PRIMARY KEY ("id"))`
    );

    await queryRunner.query(
      `ALTER TABLE "threesixty_view" ADD CONSTRAINT "FK_21baf9d82d8411eebe560242ac1" FOREIGN KEY ("product_id") REFERENCES "product"("id") ON DELETE CASCADE ON UPDATE NO ACTION`
    );
}
```

I have scoured through multiple threads on this topic over here, discord, and seen the official medusa-plugin-restock-notification plugin in the medusajs github repository and I cant seem to get why the main backed is not picking up the relationship properly. 

Any help on this please?

**Comment by sheikhumar93** [comment]
I have a `datasource.js` file in my medusa backend

```
entities: [
    "dist/models/.js",
    "node_modules/@medusajs/medusa/dist/models/*.js",
    "../packages/medusa-plugin-interactive-360/dist/models/*.js",
  ],
  migrations: [
    "dist/migrations/.js",
    "node_modules/@medusajs/medusa/dist/migrations/*.js",
    "../packages/medusa-plugin-interactive-360/dist/migrations/*.js",
  ],
```

and in plugin folder

```
entities: [
    "dist/models/.js",
    "node_modules/@medusajs/medusa/dist/models/*.js",
  ],
  migrations: [
    "dist/migrations/.js",
    "node_modules/@medusajs/medusa/dist/migrations/*.js",
  ],
```

but still the error continues to occur

**Comment by diegonc** [comment]
I had the same problem and it's because there are multiple `@medusajs/medusa` packages installed, one from the project and one from the plugin. This confuses typeorm.

My solution was to set a peerDependency and a devDependency on medusa package. The devDependency allows you to run the build step and the peerDependency (actually not being a normal dependency) makes npm omit the package when installing on production (like when publishing).

Then, publish the plugin. The `npm link` instructions didn't work for me as I couldn't avoid the problem in the first line of this post :/

**Comment by radoslaw-sz** [comment]
hi @sheikhumar93  and @diegonc 
I have a similar issue. I have developed a plugin, published to npm and I want now use it in medusa.
I used "npm install plugin-name", it has been installed correctly, everything is built.
But when "npm run start" I have a similar issue as mentioned above.

Any clue how to avoid? Is this common for all plugins or I am doing something wrong?

**Reply by radoslaw-sz** [reply]
Nevermind, I have removed it from node_modules from plugin and it works so I guess this peerDependency/devDependency will work.
edit: it works.
